### PR TITLE
feat(core): allow fully customizable indices settings in `snowowl.yml` configuration file

### PR DESCRIPTION
**Pull request by cmark**
Each index can receive its own custom settings object nested under `repository.index` via its index name. If not specified, index
initialization will fall back to the default values.

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/TerminologyRepository.java [hunk 1]
```diff
@@ -83,9 +83,13 @@ public String id() {
...
 		
-		final Map<String, Object> indexSettings = newHashMap(context.service(IndexSettings.class));
-		indexSettings.put(IndexClientFactory.NUMBER_OF_SHARDS, context.service(SnowOwlConfiguration.class).getModuleConfig(RepositoryConfiguration.class).getIndexConfiguration().getNumberOfShards());
-		final IndexClient indexClient = Indexes.createIndexClient(repositoryId, mapper, mappings, indexSettings);
+		IndexConfiguration indexConfiguration = context.service(RepositoryConfiguration.class).getIndexConfiguration();
+		final IndexClient indexClient = Indexes.createIndexClient(
+			repositoryId, 
+			mapper, 
+			mappings, 
+			context.service(IndexSettings.class).forIndex(indexConfiguration, repositoryId)
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/TerminologyRepository.java:91]: Same for `Indexes.createIndexClient`, although the client name here is already parameterized.

File: cis/com.b2international.snowowl.snomed.cis/src/com/b2international/snowowl/snomed/cis/SnomedIdentifierPlugin.java [hunk 2]
```diff
@@ -116,7 +117,12 @@ private void registerSnomedIdentifierService(final SnomedIdentifierConfiguration
 
 		switch (conf.getStrategy()) {
 		case EMBEDDED:
-			final Index index = Indexes.createIndex(SNOMED_IDS_INDEX, env.service(ObjectMapper.class), new Mappings(SctId.class), env.service(IndexSettings.class));
+			final Index index = Indexes.createIndex(
+				SNOMED_IDS_INDEX, 
+				env.service(ObjectMapper.class), 
+				new Mappings(SctId.class), 
+				env.service(IndexSettings.class).forIndex(env.service(RepositoryConfiguration.class).getIndexConfiguration(), SNOMED_IDS_INDEX)
```
Comment (apeteri) [inline cis/com.b2international.snowowl.snomed.cis/src/com/b2international/snowowl/snomed/cis/SnomedIdentifierPlugin.java:124]: The request to get the specific index configuration could be pushed down to `Indexes.createIndex` itself, as it already knows about the index name passed in the first arg. Then you don't have to change all invocations.

**Review by apeteri — CHANGES_REQUESTED** [review]

**Review by nagyo — APPROVED** [review]
LGTM!

File: cis/com.b2international.snowowl.snomed.cis/src/com/b2international/snowowl/snomed/cis/SnomedIdentifierPlugin.java (hunk 2 above)
Comment (cmark) [reply cis/com.b2international.snowowl.snomed.cis/src/com/b2international/snowowl/snomed/cis/SnomedIdentifierPlugin.java:124]: The problem is a reference to the `IndexConfiguration` class, which is a Snow Owl specific class with Snow Owl defaults, while the index layer is treated like a library and unfortunately it does not have access to that class.

**Review by nagyo — APPROVED** [review]
👍🏻

**Review by apeteri — APPROVED** [review]

**Pull request merged**